### PR TITLE
Add function to collect all of a file's transitive includes

### PR DESCRIPTION
**Pull request by begroh**
**Describe your changes**
Add a function to IncludeGraphDependencies to collect all of a file's direct and intransitive includes.

**Testing performed**
Added unit tests to verify expected behavior.

### CODE REVIEW

**Comment by begroh** [comment]
@dbeer1 @envp @HaltCatchFire

File: include/clangmetatool/include_graph_dependencies.h [hunk 1]
```diff
@@ -22,6 +23,16 @@ struct IncludeGraphDependencies {
   decrementUsageRefCount(clangmetatool::collectors::IncludeGraphData *data,
                          const clangmetatool::types::FileGraphEdge &edge);
 
+
+  /**
+   * Collect all direct and transitive includes of a file without checking
+   * whether or not the initial file actually depends on them.
+   */
+  static std::unordered_set<clangmetatool::types::FileUID>
```
Comment (dbeer1) [inline include/clangmetatool/include_graph_dependencies.h:31]: Lets make this std::set for uniformity with the existing interface?

File: t/034-validate-reference-count.t.cpp [hunk 2]
```diff
@@ -0,0 +1,188 @@
+#include "clangmetatool-testconfig.h"
```
Comment (dbeer1) [inline t/034-validate-reference-count.t.cpp:1]: This file is not being built, and also seems to be largely a copy of `031-validate-include-graph.t.cpp`. Was it added accidentally?

Comment (begroh) [reply t/034-validate-reference-count.t.cpp:1]: Whoops, yeah, shouldn't have been committed.

**Pull request merged**